ouroboros-consensus/src/Ouroboros/Consensus/Storage/ChainDB/Impl/Background.hs

```diff
@@ -203,6 +205,12 @@ copyToImmutableDB CDB{..} = withCopyLock $ do
     mustBeUnlocked = fromMaybe
                    $ error "copyToImmutableDB running concurrently with itself"
 
+data SnapAndFlushCounters = SnapAndFlushCounters {
```

Comment:
I wonder if there are any invariants that should hold, which we could help enforce by making the constructor and fields private. Eg if we change `prevSnapshotNum` then `prevSnapshotTime` has to change as well.

Reply:
The constructor and fields are already private, they are only used in this file